src/components/organisms/EarthEditor/CanvasArea/hooks.ts

```diff
@@ -140,6 +149,39 @@ export default (isBuilt?: boolean) => {
     document.title = title;
   }, [isBuilt, title]);
 
+  const [updateLayerLatLng] = useUpdatePropertyValueLatLngMutation();
+
+  const handleDragLayer = useCallback((_layerId: string, _position: LatLngHeight | undefined) => {
+    setIsDragging(true);
+  }, []);
+
+  const handleDropLayer = useCallback(
+    async (layerId: string, position: LatLngHeight | undefined) => {
+      setIsDragging(false);
+      const layerProperty = layers?.layers.find(l => l.id === layerId)?.rawProperty;
+      const fieldId = "location";
+      const schemaItemId = "default";
+      const propertyItem =
+        layerProperty && "items" in layerProperty
+          ? layerProperty?.items.find(
+              i => i.__typename === "PropertyGroup" && i.fields.find(f => f.fieldId === "location"),
+            )
+          : undefined;
+      if (!layerProperty?.id || !position) return;
+      await updateLayerLatLng({
+        variables: {
+          propertyId: layerProperty?.id,
+          itemId: propertyItem?.id,
```

Comment:
```suggestion
          propertyId: layerProperty.id,
          itemId: propertyItem.id,
```
Since you have the check on line 170 I don't think you need the question marks here. If you get a warning here, you can just update line 70 like this:
```
if (!layerProperty || !layerProperty.id || !position) return;